src/VisualStudio/Core/Def/ValueTracking/ValueTrackingToolWindow.cs

```diff
@@ -49,28 +62,33 @@ public ValueTrackingToolWindow() : base(null)
             Content = _root;
         }
 
-        public ValueTrackingToolWindow(ValueTrackingTreeViewModel viewModel)
-            : base(null)
+        public void Initialize(ValueTrackingTreeViewModel viewModel, Workspace workspace, IThreadingContext threadingContext)
         {
-            Caption = ServicesVSResources.Value_Tracking;
-            Content = _root;
+            Contract.ThrowIfTrue(Initialized);
+
+            Initialized = true;
             ViewModel = viewModel;
+            _workspace = workspace;
+            _threadingContext = threadingContext;
+
+            _workspace.WorkspaceChanged += OnWorkspaceChanged;
         }
 
-        public TreeItemViewModel? Root
+        private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
         {
-            get => ViewModel?.Roots.Single();
-            set
-            {
-                if (value is null)
-                {
-                    return;
-                }
+            Contract.ThrowIfFalse(Initialized);
 
-                Contract.ThrowIfNull(ViewModel);
+            if (e.Kind is WorkspaceChangeKind.SolutionCleared
+                       or WorkspaceChangeKind.SolutionRemoved)
+            {
+                RoslynDebug.AssertNotNull(_threadingContext);
+                RoslynDebug.AssertNotNull(ViewModel);
```

Comment:
I'm surprised you had to say this since the Contract.ThrowIfNull() since I would have thought this was addressed by the MemberNotNull(), but maybe the compiler isn't smart enough for that. Might be worth checking with @333fred, but doesn't really impact this PR.

Reply:
This shouldn't be required.

Reply:
These were required when I had the annotation backwards. Not needed anymore, so I'll remove.